homer0/projext-plugin-rollup
# refactor: remove custom implementation of babel/polyfill

### What does this PR do?

This is the followup of homer0/projext#97 and almost a revert of #31.

When I added the custom polyfill for Babel I didn't realize that the idea of deprecating it was so the env preset could take care of adding the polyfills when they were necessary.

What I did on #31 was basically add both `core-js` and `regenerator-runtime` to the generated bundles, the entire packages... which resulted on huge builds.

On homer0/projext#97, I properly configured the env preset so it will only include the necessary polyfills, so there's no need to add the `polyfill.js` this plugin was adding; and since the core is now taking care again of the polyfills, there's no need to install `core-js` nor `regenerator-runtime` from here (and since they're not installed, this is a breaking change).

### How should it be tested manually?

First, run `projext analyze` on your target to see how much space is taking `core-js`, then... 

1. Install `projext#next` from Github and this branch.
2. Delete the projext version installed inside this plugin `node_modules` (since this is not installed from a "proper version", `npm`/`yarn`, may install the one required by the plugin inside its directory: `node_modules/projext-plugin-rollup/node_modules/projext`).

Now, run `analyze` again and `core-js` should be "smaller" than before, as now it's only including what it's actually needed.

Also, don't forget to...

```bash
npm test
# or
yarn test
```
